stores/src/main/java/com/issoft/training/stores/listCategories/Store.java

```diff
@@ -23,4 +28,40 @@ public void setCategoryList(List<Category> categoryList) {
     public void printListCategories() {
         System.out.println(categoryList);
     }
+
+    private List<Product> getAllShopProducts() {
+        List<Product> allProductslist = new ArrayList<>();
+        for (Category cat : categoryList) {
+            for (Product pro : cat.getListProduct()) {
+//                System.out.println(pro);
+                allProductslist.add(pro);
+            }
+        }
+        return allProductslist;
+    }
+
+    public void sortProducts() {
+        DocPars doc = new DocPars();
+        doc.DocParse("config.xml");
+        String orderingName = doc.getOrderingValueByTag("name");
+        String orderingPrice = doc.getOrderingValueByTag("price");
+        String orderingRate = doc.getOrderingValueByTag("rate");
+        List<Product> listNewProduct = new ArrayList<>(getAllShopProducts());
```

Comment:
why do you not return new list from getAllShop...method, just not to write this line each time.